components/hab/src/command/studio/enter.rs

```diff
@@ -240,13 +240,20 @@ mod inner {
             return false;
         }
 
+        for arg in args.iter() {
+            let str_arg = arg.to_string_lossy();
+            if str_arg == String::from("-D") {
+                return false;
+            }
+        }
+
         for arg in args.iter() {
```

Comment:
A comment to point out that this particular chunk of code is in support of a deprecated flag, and thus fair-game for removal soon would be helpful.